tkg/clusterclient/clusterclient.go

```diff
@@ -2055,6 +2078,34 @@ func (c *client) PatchImageRepositoryInKubeProxyDaemonSet(newImageRepository str
 	return nil
 }
 
+// PatchKappControllerLastAppliedAnnotation ensures the kapp-controller deployment on the cluster has last-applied annotation
+func (c *client) PatchKappControllerLastAppliedAnnotation(namespace string) error {
```

Comment:
lets write a unit test for this

Reply:
Thanks Vijay for taking a look. I tried but seems we can't simulate terminal `kubectl` behavior in the unit tests. Some other similar functions like `kubectlApply` and `kubectlGetResource` don't have corresponding tests too. But tests for the addon upgrade flow for kapp-controller has been added.